docs/commands_states.rst

```diff
@@ -474,6 +474,37 @@ between 2015 and 2022 with MON windows commanded::
    CCD temperatures are set to -20 C and the ``.acqs`` and ``.guides`` attributes
    are stubbed with empty tables.
 
+Getting dicts instead of ACA tables
+"""""""""""""""""""""""""""""""""""
+Another performance option which can be useful in some cases is to set the
+``as_dict`` keyword to ``True``. This will return a list of dictionaries instead
+of converting each catalog into an ``ACATable`` object.
+
+Getting a Table of catalog entries
+""""""""""""""""""""""""""""""""""
+For some use cases you want a single table of all star catalog entries matching
+the specified criteria. This can be done with the
+:func:`~kadi.commands.observations.get_starcats_as_table` function. This is
+roughly the equivalent of doing a Table ``vstack`` of the individual
+``ACATable`` catalogs but is much faster. In addition two columns ``obsid`` and
+``starcat_date`` are added to provide this information for each entry.
+
+Caching
+"""""""
+
+In order to significantly speed up the retrieval of historical star catalogs for
+typical ACA operations analysis, the results of each call to ``get_starcats()``
+are (by default) cached in a file ``~/.kadi/starcats.db``.
+
+This caching is controlled by a configuration parameter ``cache_starcats``. To
+permanently disable caching you can edit your configuration file (see
+:ref:`configuration-options`). To disable caching for a single call to
+``get_starcats()``, you can do something like::
+
+    >>> from kadi.commands import get_starcats, conf
+    >>> with conf.set_temp('cache_starcats', False):
```

Comment:
Somewhat surprised to get different results with use of as_dict and with and without cache.
```
In [4]: with conf.set_temp('cache_starcats', False):
   ...:     print(kadi.commands.get_starcats(obsid=23800, set_ids=False, as_dict=True))
   ...: 
[<ACATable length=11>
 slot  idx  type  sz   maxmag   yang     zang    dim   res  halfw
int64 int64 str3 str3 float64 float64  float64  int64 int64 int64
----- ----- ---- ---- ------- -------- -------- ----- ----- -----
    0     1  FID  8x8    8.00  -773.14 -1741.99     1     1    25
    1     2  FID  8x8    8.00  2140.38   166.73     1     1    25
    2     3  FID  8x8    8.00 -1826.24   160.26     1     1    25
    3     4  BOT  6x6    8.70  -277.35  1178.46    28     1   160
    4     5  BOT  6x6    9.84 -1329.24 -2103.75    28     1   160
    5     6  BOT  6x6   10.58  1758.15  -270.45    28     1   160
    6     7  BOT  6x6   11.02   761.76  1753.00    28     1   160
    7     8  BOT  6x6   10.03  1349.94  -786.44    28     1   160
    0     9  ACQ  6x6   11.08  1777.10   544.73    28     1   160
    1    10  ACQ  6x6   11.08 -1534.55 -1989.80    16     1   100
    2    11  ACQ  6x6   11.14   647.71  -848.31    24     1   140]

In [5]: with conf.set_temp('cache_starcats', True):
   ...:     print(kadi.commands.get_starcats(obsid=23800, set_ids=False, as_dict=True))
   ...: 
[{'slot': [0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2], 'idx': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], 'id': [2, 4, 5, 29886088, 29887352, 30288600, 30290968, 30292744, 30289640, 29892168, 29884784], 'type': ['FID', 'FID', 'FID', 'BOT', 'BOT', 'BOT', 'BOT', 'BOT', 'ACQ', 'ACQ', 'ACQ'], 'sz': ['8x8', '8x8', '8x8', '6x6', '6x6', '6x6', '6x6', '6x6', '6x6', '6x6', '6x6'], 'mag': [7.0, 7.0, 7.0, 7.197758197784424, 8.239758491516113, 9.071069717407227, 9.502711296081543, 8.467659950256348, 9.440195083618164, 9.462800025939941, 9.66943359375], 'maxmag': [8.0, 8.0, 8.0, 8.703125, 9.84375, 10.578125, 11.015625, 10.03125, 11.078125, 11.078125, 11.140625], 'yang': [-773.1356538419711, 2140.3768711267426, -1826.2357244982522, -277.3544147694628, -1329.2356648487812, 1758.1517798656714, 761.758058124696, 1349.939334260924, 1777.1017768640722, -1534.5544169835614, 647.7143148477992], 'zang': [-1741.9918297658562, 166.72682929256607, 160.26432841548214, 1178.4580649819864, -2103.748188777892, -270.4481650397396, 1753.0017958871406, -786.4356810063402, 544.7330767879428, -1989.7980937978184, -848.3106475100061], 'dim': [1, 1, 1, 28, 28, 28, 28, 28, 28, 16, 24], 'res': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 'halfw': [25, 25, 25, 160, 160, 160, 160, 160, 160, 100, 140], 'meta': OrderedDict([('obsid', 23800), ('att', <Quat q1=0.17787506 q2=0.45262507 q3=0.82743652 q4=0.28078429>), ('date', '2022:001:17:00:58.521'), ('duration', 28259.74899995327), ('t_ccd_guide', -20.0), ('t_ccd_acq', -20.0), ('detector', 'ACIS-S'), ('sim_offset', 4)])}]
```

Reply:
This appears to be fixed with 35fd740, please test.

Reply:
Tested.